### PR TITLE
Fix flaky metrics, health, and observability tests

### LINKED ISSUE #218: [flaky] pkg/channel/distributed/common/metrics: TestMetricsServer_Report
https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka/132/pull-knative-sandbox-eventing-kafka-unit-tests/1330891148914331648
```
=== RUN   TestMetricsServer_Report
    logger.go:130: 2020-11-23T15:12:34.872Z	INFO	profiling/server.go:59	Profiling enabled: false
    logger.go:130: 2020-11-23T15:12:34.887Z	INFO	metrics/exporter.go:160	Flushing the existing exporter before setting up the new exporter.
    logger.go:130: 2020-11-23T15:12:34.896Z	INFO	metrics/prometheus_exporter.go:51	Created Opencensus Prometheus exporter with config: &{eventing-kafka eventing_kafka prometheus 5000000000 <nil> <nil>  false 9878 0.0.0.0 false   {   false}}. Start the server for Prometheus exporter.
    logger.go:130: 2020-11-23T15:12:34.896Z	INFO	metrics/exporter.go:173	Successfully updated the metrics exporter; old config: <nil>; new config &{eventing-kafka eventing_kafka prometheus 5000000000 <nil> <nil>  false 9878 0.0.0.0 false   {   false}}
    stats_reporter_test.go:103: 
        	Error Trace:	stats_reporter_test.go:103
        	Error:      	Should be true
        	Test:       	TestMetricsServer_Report
--- FAIL: TestMetricsServer_Report (0.21s)
```

### PR DESCRIPTION
**Pull request by eric-sap**
Fixes #218 

## Proposed Changes

- 🐛 The "live server" tests are not necessary for decent code coverage during unit testing; replaced with mocks

### CODE REVIEW

File: pkg/channel/distributed/common/config/observability.go [hunk 1]
```diff
@@ -32,6 +34,10 @@ import (
 	"knative.dev/pkg/profiling"
 )
 
+var ListenAndServeWrapper = func(srv *nethttp.Server) func() error { return srv.ListenAndServe }
+var StartWatcherWrapper = func(cmw *configmap.InformedWatcher, done <-chan struct{}) error { return cmw.Start(done) }
+var UpdateExporterWrapper = metrics.UpdateExporter
+
```
Comment (travis-minke-sap) [inline pkg/channel/distributed/common/config/observability.go:40]: can we put into a single `var` block and add a comment saying these are to facilitate unit testing?

Comment (eric-sap) [reply pkg/channel/distributed/common/config/observability.go:44]: Yes; seems like a good idea.

**Pull request merged**